### PR TITLE
feat: Admin V2 Customers

### PR DESCRIPTION
**Pull request by olivermrbl**
Extracted almost all of it from #6932 and then cleaned it up and added the new hooks

### CODE REVIEW

File: packages/admin-next/dashboard/src/v2-routes/customers/customer-detail/components/customer-group-section/customer-group-section.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// TODO: Should be added with Customer Groups UI
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/v2-routes/customers/customer-detail/components/customer-group-section/customer-group-section.tsx:1]: Should be added back when adding in Customer Groups UI

File: packages/admin-next/dashboard/src/v2-routes/customers/customer-create/components/create-customer-form/create-customer-form.tsx [hunk 2]
```diff
@@ -1,61 +1,44 @@
...
-        path: ["password_confirmation"],
-      })
-    }
-  })
+const CreateCustomerSchema = zod.object({
+  email: zod.string().email(),
+  first_name: zod.string().min(1),
+  last_name: zod.string().min(1),
+  phone: zod.string().min(1).optional(),
+})
```
Comment (sradevski) [inline packages/admin-next/dashboard/src/v2-routes/customers/customer-create/components/create-customer-form/create-customer-form.tsx:18]: How would passwords be handled now? Send a reset password link?

File: packages/types/src/http/customer/admin/customer.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * @experimental
+ */
+export interface AdminCustomerResponse {
+  customer: CustomerResponse
+}
+
+/**
+ * @experimental
```
Comment (sradevski) [inline packages/types/src/http/customer/admin/customer.ts:65]: What's the purpose of `experimental`?

**Review by sradevski — APPROVED** [review]
Looking good!

File: packages/types/src/http/customer/admin/customer.ts (hunk 3 above)
Comment (olivermrbl) [reply packages/types/src/http/customer/admin/customer.ts:65]: 1.0 and 2.0 types are exported from the same package, so the @experimental flag was added so external developers don't confuse the types.

File: packages/admin-next/dashboard/src/v2-routes/customers/customer-create/components/create-customer-form/create-customer-form.tsx (hunk 2 above)
Comment (olivermrbl) [reply packages/admin-next/dashboard/src/v2-routes/customers/customer-create/components/create-customer-form/create-customer-form.tsx:18]: Yes, we should not be able to set the password on customer accounts. That is a separate flow after creating the customer.

**Pull request merged**